openssl/src/bn/mod.rs

```diff
@@ -102,6 +102,18 @@ impl BigNum {
         })
     }
 
+    pub unsafe fn new_from_ffi(orig: *mut ffi::BIGNUM) -> Result<BigNum, SslError> {
+        if orig.is_null() {
+            panic!("Null Pointer was supplied to BigNum::new_from_ffi");
+        }
+        let r = ffi::BN_dup(orig);
+        if r.is_null() {
+            panic!("Unexpected null pointer from BN_dup(..)")
```

Comment:
`SslError::get()` will return the contents of the error stack that OpenSSL will set when this returns NULL.